src/vstart.sh

```diff
@@ -692,6 +692,9 @@ EOF
 EOF
             echo adding osd$osd key to auth repository
             ceph_adm -i "$key_fn" auth add osd.$osd osd "allow *" mon "allow profile osd" mgr "allow profile osd"
+            if [ "$spdk_enabled" -eq 1 ]; then
+                sleep 3 # wait for NVME controller initialization ready
```

Comment:
could you remind me where NVMe controller is initialized?

Reply:
Hi Kefu,

The time out happened in nvme_ctrlr_process_init() in spdk/lib/nvme/nvme_ctrlr.c.

Br,
Tone

Reply:
@tchaikov Thanks for your comments. The time out is because the NVMe device is not ready in time. I observed the problem on both x86_64 and aarch64 platforms with one NVMe card for each test bed.

With the workaround I can  start up Ceph cluster conveniently. ;)

i will try SPDK/fio_plugin and share the state with you.

Thanks!

Reply:
@tchaikov Kefu, as the quick check, with the SPDK used in Ceph master/LATEST,  I cannot launch fio. It is not time out error, but DMA memory mapping error. After switch to SPDK mater/LATEST version, I can pass the fio test.

For the short delay in vstart.sh, it should be placed ahead the starting of the second "ceph-osd". I will update the change soon.

In the current SPDK code used in Ceph, the above issues are fixed.

I ran the simple test to verify the code "https://github.com/spdk/spdk/blob/2772d86c23148d3cd575b1c814584fa0946acc0d/lib/nvme/nvme_ctrlr.c#L1951-L1961", it seems the “spdk_deply_us(100)” is NOT enough. After extend the delay to 1000000, I did not see the "time out" error during my several trials. II will leave message in https://github.com/spdk/spdk/issues/139.

I think the short delay in vstart.sh can resolve the issue although it is SPDK issue. Anyhow the developer can set up the development Ceph cluster enabling SPDK with the change. What do you think? :)

Thanks!